src/plotman/graph.py

```diff
@@ -0,0 +1,263 @@
+        window_list = []
+        # This takes care of when we dont have a full window
+        for i in range(window):
+            rate_list.append(data[i] - data[0])
+            window_list.append(i)
+        # This takes care of when we do
+        for i in range(len(data) - window):
+            rate_list.append(data[i+window] - data[i])
+            window_list.append(window)
+        rate_list, window_list = np.array(rate_list), np.array(window_list)
+        rate_list[rate_list == 0] = np.nan # This prevents div by zero error
+        return np.where(np.logical_not(np.isnan(rate_list)), (window_list-1) / rate_list, 0)
+
+    # Estimate the rate of ending or the rate of starting
+    if end:
+        rate = estimate_rate(data[:,1], window)
+        ax.plot(data[:,1], rate)
+    else:
+        rate = estimate_rate(data[:,0], window)
+        ax.plot(data[:,0], rate)
+
+    ax.set_ylabel('Avg Plot Rate (plots/hour)')
+    ax.set_xlim(np.min(data[:,0])-2, np.max(data[:,1])+2)
+
+
+def create_ax_plottime(ax, data, window=3):
+    '''
+        Create a plot showing the average time to create a single plot. This is
+            computed using a moving average. Note that the plot may not be
+            very accurate for the beginning and ending windows.
+        Parameters:
+            ax: a matplotlib axis.
+            data: numpy arrary with [start times, end times].
+            window: Window to compute rate over.
+    '''
+
+    # Compute moving avg
+    kernel = np.ones(window) / window
+    data_tiled = np.vstack((
+        np.expand_dims(data[:,1] - data[:,0], axis=1),
+        np.tile(data[-1,1] - data[-1,0], (window-1, 1))
+    ))
+    rolling_avg = np.convolve(data_tiled.squeeze(), kernel, mode='valid')
+
+    ax.plot(data[:,1], rolling_avg)
+
+    ax.set_ylabel('Avg Plot Time (hours)')
+    ax.set_xlim(np.min(data[:,0])-2, np.max(data[:,1])+2)
+
+
+def create_ax_plotcumulative(ax, data):
+    '''
+        Create a plot showing the cumulative number of plots over time.
+        Parameters:
+            ax: a matplotlib axis.
+            data: numpy arrary with [start times, end times].
+    '''
+    ax.plot(data[:,1], range(data.shape[0]))
+
+    ax.set_ylabel('Total plots (plots)')
+    ax.set_xlim(np.min(data[:,0])-2, np.max(data[:,1])+2)
+
+
+def graph(logfilenames, figfile, bytmp, bybitfield):
+    data = {}
+    logfilenames = [os.path.join(os.path.dirname(logfilenames), l) for l in os.listdir(logfilenames) if
+        os.path.splitext(l)[-1] == '.log']
+
+    for logfilename in logfilenames:
+        with open(logfilename, 'r') as f:
+            # Record of slicing and data associated with the slice
+            sl = 'x'         # Slice key
+            phase_time = {}  # Map from phase index to time
+            n_sorts = 0
+            n_uniform = 0
+            is_first_last = False
+
+            # Read the logfile, triggering various behaviors on various
+            # regex matches.
+            for line in f:
```

Comment:
Do we need dedicated log parsing here?  If the existing parsing doesn't catch everything, maybe we could add the extra bits there?  Though I have to admit we have two parsers right now.  I was hoping to make some time to get everything using the newer one.  I'll try to get that done first.